storage/replica_test.go

```diff
@@ -577,6 +649,34 @@ func TestRangeTSCacheLowWaterOnLease(t *testing.T) {
 	}
 }
 
+// TestRangeLeaderLeaseRejectUnknownRaftNodeID ensures that a replica cannot
+// obtain the leader lease if it is not part of the current range descriptor.
+func TestRangeLeaderLeaseRejectUnknownRaftNodeID(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	tc.manualClock.Increment(int64(DefaultLeaderLeaseDuration + 1))
+	now := tc.clock.Now()
+	lease := &proto.Lease{
+		Start:      now,
+		Expiration: now.Add(10, 0),
+		RaftNodeID: proto.MakeRaftNodeID(2, 2),
+	}
+	args := &proto.BatchRequest{}
+	args.Add(&proto.LeaderLeaseRequest{Lease: *lease})
+	errChan, pendingCmd := tc.rng.proposeRaftCommand(tc.rng.context(), args)
```

Comment:
It's difficult because there is only one store; you would need to create another store, possibly hook it up to Raft.  If this is acceptable (but not optimal), i'd like to just add a Todo and move on for now.